src/App.jsx

```diff
@@ -1,29 +1,38 @@
-import { useEffect, useState } from 'react'
-import Header from './components/Header'
-import Tweets from './components/Tweets'
-import RightSide from './components/RightSide'
-import defaultTweets from './assets/data/tweets.js'
-import user from './assets/data/user.js'
+import { useEffect, useState, createContext } from "react";
+import Header from "./components/Header";
+import Tweets from "./components/Tweets";
+import RightSide from "./components/RightSide";
+import defaultTweets from "./assets/data/tweets.js";
+import user from "./assets/data/user.js";
 
+const TweetContents = createContext();
+const ThemeContext = createContext();
 function App() {
-    const [tweets, setTweets] = useState(defaultTweets)
-    const [theme, setTheme] = useState('light');
+  const [tweets, setTweets] = useState(defaultTweets);
+  const [theme, setTheme] = useState(localStorage.getItem("theme") || "light");
+  //if "theme" is undefined set theme to light
 
-    useEffect(() => {
-        theme === 'light'
-          ? document.body.style.backgroundColor = 'white'
-          : document.body.style.backgroundColor = 'black'
-    }, [theme])
+  localStorage.setItem("theme", theme);
 
-    return (
+  useEffect(() => {
+    theme === "light"
```

Comment:
We can write this a bit more efficiently. Ternary statements return a value. So we can write it like this
```js
 const style = theme === "light"
      ? "white" : "black";
 document.body.style.backgroundColor = style
```